server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -1699,7 +1700,20 @@ public synchronized void close() throws IOException {
      * logic should use Node Shutdown, see {@link org.elasticsearch.cluster.metadata.NodesShutdownMetadata}.
      */
     public void prepareForClose() {
+        HttpServerTransport httpServerTransport = injector.getInstance(HttpServerTransport.class);
+        FutureTask<Void> stopper = new FutureTask<>(() -> {
+            httpServerTransport.stop();
+            return null;
+        });
+        new Thread(stopper).start();
```

Comment:
We should give the thread a name, so that a thread dump while shutting down will be understandable.

Reply:
Named `http-server-transport-stop`